datasets/switchboard.py

```diff
@@ -0,0 +1,293 @@
+            ]
+        )
+        shutil.move("ws97_speaker_stats.text", self.out_speakers_stats_file)
+
+
+class CreateSwitchboardSpeakersListJob(Job):
+    """
+    Given some speakers statistics info, this job creates a text file having on each line:
+        speaker_id gender recording
+    """
+
+    def __init__(self, speakers_stats_file):
+        """
+        :param tk.Path speakers_stats_file: speakers stats text file
+        """
+        self.speakers_stats_file = speakers_stats_file
+        self.out_speakers_list = self.output_path("speakers_list.txt")
+
+    def tasks(self):
+        yield Task("run", mini_task=True)
+
+    def run(self):
+        speaker_id, gender, rec_name = None, None, None
+        with uopen(self.speakers_stats_file) as read_f, uopen(
+            self.out_speakers_list, "w"
+        ) as out_f:
+            for line in read_f:
+                l = line.strip().split()
+                if len(l) < 2:
+                    continue
+                if l[1] == "F" or l[1] == "M":  # start new speaker
+                    speaker_id = l[0]
+                    gender = l[1]
+                    rec_name = l[2]
+                elif l[0].endswith("A") or l[0].endswith("B"):  # recording name
+                    rec_name = l[0]
+                else:
+                    continue
+
+                if speaker_id:
+                    out_f.write(
+                        speaker_id + " " + gender + " " + rec_name + "\n"
+                    )  # speaker_id gender recording
+
+
+class CreateSwitchboardBlissCorpusJob(Job):
+    """
+    Creates Switchboard bliss corpus xml
+
+    segment name format: sw2001B-ms98-a-<folder-name>
+    """
+
+    def __init__(self, audio_dir, trans_dir, speakers_list_file):
+        """
+        :param tk.Path audio_dir: path for audio data
+        :param tk.Path trans_dir: path for transcription data. see `DownloadSwitchboardTranscriptionAndDictJob`
+        :param tk.Path speakers_list_file: path to a speakers list text file with format:
+            speaker_id gender recording
+        on each line. see `CreateSwitchboardSpeakersListJob` job
+        """
+        self.audio_dir = audio_dir
+        self.trans_dir = trans_dir
+        self.speakers_list_file = speakers_list_file
+        self.out_corpus = self.output_path("swb.corpus.xml.gz")
+
+    def tasks(self):
+        yield Task("run", mini_task=True)
+
+    def run(self):
+        c = corpus.Corpus()
+        c.name = "switchboard-1"
+
+        rec_to_segs = self._get_rec_to_segs_map()
+
+        rec_to_speaker = {}
+        with uopen(self.speakers_list_file) as f:
+            for line in f:
+                l = line.strip().split()
+                assert len(l) == 3
+                assert (
+                    l[2] not in rec_to_speaker
+                ), "duplicate recording name: {}?".format(l[2])
+                assert l[1] in ["F", "M"], "Gender info must be either M or F"
```

Comment:
Lets hope no one from the LBGTQ+ community reads our code xP

Reply:
> Will you later also add jobs for hub5'00 hub5'01 etc in this file?
> What about Fisher train data?

yes Hub5'00 and Hub5'01 will be added later. I am not sure if will add Fisher.